src/shared/components/VideoTileBase/VideoTileBase.tsx

```diff
@@ -147,7 +147,8 @@ export const VideoTileBase: React.FC<VideoTileBaseProps> = ({
   onDeleteVideoClick,
   isPullupDisabled,
 }) => {
-  const { openContextMenu, contextMenuOpts } = useContextMenu()
+  const contextMenuInstanceRef = useRef<TippyInstance>()
```

Comment:
We cannot require every consumer of `ContextMenu` to maintain its own instance of tippy. Let's change the `ContextMenu` to take `items` prop, for example:
```
type ContextMenuItem = {
  icon?: React.ReactNode
  title: string
  onClick?: () => void
```

Then `onClick` handlers can be managed by `ContextMenu` itself - this way we can trigger the tippy close without bothering the consumer of `ContextMenu`

Reply:
I see what you mean, but in this case I think either way works as I don't expect `ContextMenu` to be used anywhere else than the videoTileBase.

PopoverDialog would be another discussion but I do think is needed in there for finetuning

Reply:
> I don't expect ContextMenu to be used anywhere else than the videoTileBase

I'm not so sure about it, but even if it was used only there, that's still a good reason to improve its API.